// @paperist/remark-math/lib/InlineMathTokenizer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var inlineMath_1 = require("./peg/inlineMath");
var InlineMathLocator = function (value, fromIndex) {
    var location = value.indexOf('$', fromIndex);
    if (location === -1) {
        location = value.indexOf('\\(', fromIndex);
    }
    return location;
};
var InlineMathTokenizerFunction = function (eat, value, silent) {
    var result;
    try {
        result = inlineMath_1.parse(value);
    }
    catch (err) {
        return silent ? false : undefined;
    }
    if (silent) {
        return true;
    }
    var matchStr = value.substring(result.location.start.offset, result.location.end.offset);
    var node = {
        type: 'inlineMath',
        value: matchStr,
        math: result.math,
    };
    return eat(matchStr)(node);
};
var InlineMathTokenizer = Object.assign(InlineMathTokenizerFunction, {
    locator: InlineMathLocator,
    notInBlock: true,
    notInList: true,
    notInLink: true,
});
exports.default = InlineMathTokenizer;
